tilequeue/command.py

```diff
@@ -923,6 +924,60 @@ def tilequeue_enqueue_tiles_of_interest(cfg, peripherals):
     logger.info('%d tiles of interest processed' % n_toi)
 
 
+def coord_pyramids(coords, zoom_start, zoom_stop):
+    """
+    generate full pyramid for coords
+
+    Generate the full pyrmaid for the list of coords. Note that zoom_stop is
+    exclusive.
+    """
+    for coord in coords:
+        for child_coord in coord_children_range(coord, zoom_stop):
+            if zoom_start <= child_coord.zoom < zoom_stop:
```

Comment:
Looks like `coord_children_range` doesn't yield `coord` itself. Is that intentional?

Also, looks like the yielded coords from `coord_children_range` cannot have a zoom outside this range, so is the check here superfluous?

Reply:
> Looks like coord_children_range doesn't yield coord itself. Is that intentional?

Whoops, no.

> Also, looks like the yielded coords from coord_children_range cannot have a zoom outside this range, so is the check here superfluous?

I wanted to separate the zoom at which tiles were grouped, from the zoom at which jobs were emitted. In this specific case, I wanted to group at zoom level 10, but only enqueue tiles 11-15.

I think the reason I went down that path was that we had an sqs queue for z0-10, and a different one for z11+. In retrospect though, to me it's a better idea to just alter that tier to have it be z0-9 for the low zooms, and z10+ for the high.

If that's the case, then we probably want to just remove the zoom_start/zoom_stop command line args completely. The start zoom for the pyramid would just be the group by zoom from the config, and the stop zoom can be read from `cfg.max-zoom`. This will simplify the code and is probably always what we want. What do you think?

Reply:
Ah, I get it now. The `child_coord.zoom` might be less than `start_zoom`. But I still don't see how it can be greater than or equal to `stop_zoom`.

I think it makes a lot of sense to enqueue whole pyramids. But I'm not sure that we'll always want that. Perhaps just default `start_zoom` to the group zoom and `stop_zoom` to the max zoom, but still allow them to be overridden?

Reply:
> But I still don't see how it can be greater than or equal to stop_zoom.

You're right, it should never be the case. I can remove that part.

Reply:
> Looks like coord_children_range doesn't yield coord itself. Is that intentional?

Fixed in 91804bd.

> But I still don't see how it can be greater than or equal to stop_zoom.

Fixed in b22856e.

> Perhaps just default start_zoom to the group zoom and stop_zoom to the max zoom, but still allow them to be overridden?

Good idea, updated in 216f95f.